### PR TITLE
Liked watch later history

### PR DESCRIPTION
**Pull request by abhishektanwar**
FEATURES ADDED:
- like and unlike video
- add and remove video to watch later
- add ,remove video from history and clear watch history

FILES TO CHECK:
- src/Contexts/PlaylistContext/index.jsx
- src/Hooks/usePlaylistApiCalls.js
- src/Reducers/userVideoDataReducer.js
- src/components/VideoCard/VideoCardDropDownMenu.jsx
- src/components/VideoCard/index.jsx
- src/screens/History/index.jsx
- src/screens/LikedVideos/index.jsx
- src/screens/WatchLater/index.jsx

### CODE REVIEW

File: src/Reducers/userVideoDataReducer.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { dispatchActioTypes } from "./dispatchActionTypes"
 
 const userVideoDataReducer = (state,action) => {
-  const {CREATE_PLAYLIST_WITH_VIDEO, UPDATE_PLAYLIST,UPDATE_LIKED_VIDEO} = dispatchActioTypes;
+  const {CREATE_PLAYLIST_WITH_VIDEO, UPDATE_PLAYLIST,UPDATE_LIKED_VIDEO,UPDATE_WATCH_LATER,UPDATE_HISTORY} = dispatchActioTypes;
   switch(action.type){
     case CREATE_PLAYLIST_WITH_VIDEO:
       // const playlists =
```
Comment (Shubhamkr2610) [inline src/Reducers/userVideoDataReducer.js:7]: remove unused code

**Review by Shubhamkr2610 — APPROVED** [review]
good to merge

File: src/Contexts/PlaylistContext/index.jsx [hunk 2]
```diff
@@ -17,6 +21,116 @@ const PlaylistProvider = ({ children }) => {
...
+          });
+          if (res.status === 200) {
+            userVideoDispatch({
+              type: UPDATE_HISTORY,
+              payload: { data:res.data },
+            });
+          }
+        }
+        catch(err){
+          console.log("initial hitory data fetch error",err)
```
Comment (iRohitGaur) [inline src/Contexts/PlaylistContext/index.jsx:50]: suggestion: remove `console.log`

File: src/components/VideoCard/index.jsx [hunk 3]
```diff
@@ -77,7 +127,13 @@ const VideoCard = (props) => {
   } else {
     return (
       <div class="horizontal-card margin-trb-20">
-        <div class="image-container badge-container cursor-pointer" onClick={() => navigate(`/video/${youtubeID}`)}>
+        <div
+          class="image-container badge-container cursor-pointer"
+          onClick={() => {
+            addVideoToHistory(user.encodedToken, video);
+            navigate(`/video/${youtubeID}`);
+          }}
```
Comment (iRohitGaur) [inline src/components/VideoCard/index.jsx:135]: suggestion: make a handler function to extract code logic from UI

**Review by iRohitGaur — APPROVED** [review]
Nice work!

Code is easy to understand.

Good to merge 👍🏻

File: src/Contexts/PlaylistContext/index.jsx (hunk 2 above)
Comment (abhishektanwar) [reply src/Contexts/PlaylistContext/index.jsx:50]: yes . Will add toasts to make user experience more better

File: src/components/VideoCard/index.jsx (hunk 3 above)
Comment (abhishektanwar) [reply src/components/VideoCard/index.jsx:135]: Sure . will look into it.

**Pull request merged**